mdal/frmts/mdal_ply.cpp

```diff
@@ -60,275 +65,190 @@ bool MDAL::DriverPly::canReadMesh( const std::string &uri )
+  if ( MDAL::Log::getLastStatus() != MDAL_Status::None ) { return nullptr; }
+  const libply::ElementsDefinition &definitions = file.definitions();
+  const libply::Metadata &metadata = file.metadata();
+  for ( const libply::Element &element : definitions )
   {
-    MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " does not contain any definitions" );
-    return nullptr;
-  }
-  chunks = split( line, ' ' );
-
-  // configure the vectors to hold the data
-
-  for ( size_t elid = 0; elid < elements.size(); ++elid )
-  {
-    MDAL::DriverPly::element element = elements[elid];
     if ( element.name == "vertex" )
     {
       vertexCount = element.size;
-      vertices.resize( vertexCount );
-      for ( size_t i = 0; i < element.properties.size(); ++i )
-      {
-        if ( element.properties[i] != "x" &&
-             element.properties[i] != "y" &&
-             element.properties[i] != "z" &&
-             !element.list[i] )
-        {
-          vertexDatasets.push_back( * new std::vector<double> );
-          vProp2Ds.push_back( element.properties[i] );
-        }
-      }
     }
     else if ( element.name == "face" )
     {
       faceCount = element.size;
-      faces.resize( faceCount );
-      for ( size_t i = 0; i < element.properties.size(); ++i )
-      {
-        if ( element.properties[i] != "vertex_indices" &&
-             !element.list[i] )
-        {
-          faceDatasets.push_back( * new std::vector<double> );
-          fProp2Ds.push_back( element.properties[i] );
-        }
-      }
     }
     else if ( element.name == "edge" )
     {
       edgeCount = element.size;
-      edges.resize( edgeCount );
-      for ( size_t i = 0; i < element.properties.size(); ++i )
-      {
-        if ( element.properties[i] != "vertex1" &&
-             element.properties[i] != "vertex2" &&
-             !element.list[i] )
-        {
-          edgeDatasets.push_back( * new std::vector<double> );
-          eProp2Ds.push_back( element.properties[i] );
-        }
-      }
     }
-
-    // load the data
-    for ( size_t i = 0; i < element.size; ++i )
+    for ( const libply::Property &property : element.properties )
     {
-      /*
-      * set the line size - we will only deal with one list at the begining of the line
-      */
-      size_t nChunks = element.properties.size();
-      if ( element.list[0] ) nChunks += MDAL::toSizeT( chunks[0] );
-      if ( chunks.size() != nChunks )
+      if ( element.name == "vertex" &&
+           property.name != "X" &&
+           property.name != "x" &&
+           property.name != "Y" &&
+           property.name != "y" &&
+           property.name != "Z" &&
+           property.name != "z"
+         )
       {
-        MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " contains invalid line : " + line );
-        return nullptr;
+        vProp2Ds.push_back( property.name );
+        vertexDatasets.push_back( * new std::vector<double> );
```

Comment:
I don't look at all the code, I fine this line odd.
Can you explain the logic of this? When this `std::vector<double>` will be deleted?
Maybe I am wrong, but for me here, you create a `std::vector` with a pointer to it, you de-reference the pointer with `*`, to copy the pointer object in the `vertexDatasets` container. The first created std::vector is never deleted.
Why do you not use `vertexDatasets.push_back(std::vector<double>() );`

Reply:
Despite the impression given by the diff - those lines are not changed from the previous version! 

However, since I wrote the previous version as well ...

I think you are right - I am not sure that if I were writing it now I would like that. I will have a look.

Reply:
added in ef1fec4